src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -224,9 +247,20 @@ protected internal override async Task<HttpResponseMessage> SendAsync(HttpReques
                 JSObject t = (JSObject)await response.ConfigureAwait(continueOnCapturedContext: true);
 
                 var status = new WasmFetchResponse(t, abortController, abortCts, abortRegistration);
-
                 HttpResponseMessage httpResponse = new HttpResponseMessage((HttpStatusCode)status.Status);
 
+                // Here we will set the ReasonPhrase so that it can be evaluated later.
+                // We do not have a status code but this will signal some type of what happened
+                // after interrogating the status code for success or not i.e. IsSuccessStatusCode
+                //
+                // https://developer.mozilla.org/en-US/docs/Web/API/Response/type
+                // opaqueredirect: The fetch request was made with redirect: "manual".
+                // The Response's status is 0, headers are empty, body is null and trailer is empty.
+                if (status.ResponseType == "opaqueredirect")
+                {
+                    httpResponse.SetReasonPhraseWithoutValidation(status.ResponseType);
```

Comment:
@karelz who could review this part? Should we try harder to map the result to status codes or expose that differently to the developers?

Reply:
Hello everyone.  Is there any update on this?

/cc @marek-safar @karelz @ManickaP

Reply:
Also there is a line in the PR comments about mapping a result and the approach for this PR:

> Instead of faking a 3XX status code the implementation uses the ReasonPhrase to return the fetch response type `opaqueredirect`.

Reply:
CC @dotnet/ncl

Hrm this is an interesting one. It seems like there is not a perfect solution here -- fetch API seems poorly made. ¯\\\_(ツ)_/¯

Without setting a normal redirect status code, we break compat with any existing code handling redirects.

But, it seems wrong to fake a status code when we don't know what the actual status code was. If we were to fake a status code, a temporary redirect seems like the best choice -- permanent redirect could cause dangerous handling e.g. updating a URI in a database.

To add a third bad option, what do we think of this? It at least gives the user some flexibility here.

```c#
public class BrowserHttpHandler
{
    public const HttpStatusCode OpaqueRedirect = (HttpStatusCode)1000;

    // if false, return TemporaryRedirect. if true, return OpaqueRedirect.
    public bool ReturnOpaqueRedirect { get; set; } = false;
}
```

(note, using a status code like this may not be advisable as an on-by-default option, as you might have some code doing e.g. `if(StatusCode >= 500)`)

Reply:
If I am understanding the above wouldn't this change the Public API as well?